openquake/hazardlib/tests/gsim/gmpe_tables_test.py

```diff
@@ -0,0 +1,753 @@
+        """
+        Tests missing period information
+        """
+        with self.assertRaises(ValueError) as ve:
+            gsim = GMPETable(self.TABLE_FILE)
+        self.assertEqual(str(ve.exception),
+                         "Spectral Acceleration must be accompanied by periods"
+                         )
+
+
+class GSIMTableTestCaseNoAmplification(unittest.TestCase):
+    """
+    Tests the simple case in which no amplification is applied
+    """
+    TABLE_FILE = os.path.join(BASE_DATA_PATH,
+                              "good_dummy_table_noamp.hdf5")
+
+    def test_instantiation(self):
+        """
+        Tests instantiation without amplification
+        """
+        gsim = GMPETable(self.TABLE_FILE)
+        self.assertIsNone(gsim.amplification)
+        self.assertSetEqual(gsim.REQUIRES_SITES_PARAMETERS, set(()))
+        self.assertSetEqual(gsim.REQUIRES_RUPTURE_PARAMETERS, set(("mag",)))
+
+    def test_get_mean_and_stddevs(self):
+        """
+        Tests mean and standard deviations without amplification
+        """
+        gsim = GMPETable(self.TABLE_FILE)
+        rctx = RuptureContext()
+        rctx.mag = 6.0
+        dctx = DistancesContext()
+        # Test values at the given distances and those outside range
+        dctx.rjb = np.array([0.5, 1.0, 10.0, 100.0, 500.0])
+        sctx = SitesContext()
+        stddevs = [const.StdDev.TOTAL]
+        expected_mean = np.array([2.0, 2.0, 1.0, 0.5, 1.0E-20])
+        # PGA
+        mean, sigma = gsim.get_mean_and_stddevs(sctx, rctx, dctx,
+                                                imt_module.PGA(),
+                                                stddevs)
+        np.testing.assert_array_almost_equal(np.exp(mean), expected_mean, 5)
+        np.testing.assert_array_almost_equal(sigma[0], 0.5 * np.ones(5), 5)
+        # SA
+        mean, sigma = gsim.get_mean_and_stddevs(sctx, rctx, dctx,
+                                                imt_module.SA(1.0),
+                                                stddevs)
+        np.testing.assert_array_almost_equal(np.exp(mean), expected_mean, 5)
+        np.testing.assert_array_almost_equal(sigma[0], 0.8 * np.ones(5), 5)
+        # PGV
+        mean, sigma = gsim.get_mean_and_stddevs(sctx, rctx, dctx,
+                                                imt_module.PGV(),
+                                                stddevs)
+        np.testing.assert_array_almost_equal(np.exp(mean),
+                                             10. * expected_mean,
+                                             5)
+        np.testing.assert_array_almost_equal(sigma[0], 0.5 * np.ones(5), 5)
+
+
+class GSIMTableQATestCase(BaseGSIMTestCase):
+    """
+    Quality Assurance test case with real data taken from the
+    2015 Canadian National Seismic Hazard Map
+    """
+    GSIM_CLASS = GMPETable
+    MEAN_FILE = "GSIMTABLES/Wcrust_rjb_med_MEAN.csv"
+    STD_TOTAL_FILE = "GSIMTABLES/Wcrust_rjb_med_TOTAL.csv"
+
```

Comment:
BTW, an all uppercase directory name (`GSIMTABLES`) looks strange to me. Do we have a convention on that? If not, I would prefer to use an all lowercase name (`gsimtables`).